src/System.Private.CoreLib/shared/System/Collections/Generic/NonRandomizedStringEqualityComparer.cs

```diff
@@ -31,4 +31,52 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
             info.SetType(typeof(GenericEqualityComparer<string>));
         }
     }
+
+    [Serializable]
+    internal sealed class NonRandomizedStringComparer : IEqualityComparer<string>, IComparer<string>, ISerializable
+    {
+        internal static IEqualityComparer<string> Default { get; } = new NonRandomizedStringComparer();
+
+        private NonRandomizedStringComparer() { }
+
+        public bool Equals(string x, string y) => string.Equals(x, y);
+
+        public int GetHashCode(string obj) => obj?.GetNonRandomizedHashCode() ?? 0;
+
+        public int Compare(string x, string y)
+        {
+            if (ReferenceEquals(x, y))
+                return 0;
+            if (x == null)
+                return -1;
+            if (y == null)
+                return 1;
+
+            return string.CompareOrdinal(x, y);
+        }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.SetType(typeof(OrdinalComparer));
+            info.AddValue("_ignoreCase", false);
```

Comment:
We are (inconsistently) adding this valuable comment:
```
 info.AddValue(...)   // Do not rename (binary serialization)
```

Reply:
Added